File: android_app/app/src/main/java/com/oceansentinels/app/presentation/ui/screens/auth/LoginScreen.kt
package com.oceansentinels.app.presentation.ui.screens.auth

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.oceansentinels.app.domain.model.UserRole
import com.oceansentinels.app.presentation.ui.components.*
import com.oceansentinels.app.presentation.ui.theme.OceanColors
import com.oceansentinels.app.presentation.viewmodel.AuthViewModel
import com.oceansentinels.app.presentation.viewmodel.LoginState

/**
 * Login screen - matches app1 Figma design
 * Yellow header with "Ocean Sentinels Portal", gray input fields, yellow Sign In button
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LoginScreen(
    onNavigateToRegister: () -> Unit,
    onNavigateToHome: () -> Unit,
    viewModel: AuthViewModel = hiltViewModel()
) {
    val loginState by viewModel.loginState.collectAsState()
    
    var username by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var passwordVisible by remember { mutableStateOf(false) }
    var rememberMe by remember { mutableStateOf(false) }
    var showDemoAccounts by remember { mutableStateOf(false) }
    
    // Handle login success
    LaunchedEffect(loginState) {
        if (loginState is LoginState.Success) {
            onNavigateToHome()
            viewModel.resetLoginState()
        }
    }
    
    val scrollState = rememberScrollState()
    
    Box(modifier = Modifier
        .fillMaxSize()
        .background(MaterialTheme.colorScheme.background)
        .navigationBarsPadding()
    ) {
        // Ocean watermark background - subtle blue circle
        Box(
            modifier = Modifier
                .size(400.dp)
                .offset(x = (-100).dp, y = 60.dp)
                .alpha(0.08f)
                .background(OceanColors.Secondary, shape = RoundedCornerShape(50))
        )
        
        Column(
            modifier = Modifier.fillMaxSize()
        ) {
            // Yellow header bar matching Figma design
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(OceanColors.Primary)
                    .statusBarsPadding()
                    .padding(horizontal = 24.dp, vertical = 16.dp)
                    .height(120.dp),
                contentAlignment = Alignment.BottomStart
            ) {
                Column {
                    Text(
                        text = "Ocean Sentinels Portal",
                        style = MaterialTheme.typography.headlineSmall,
                        color = MaterialTheme.colorScheme.onPrimary
                    )
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = "Secure access to India's coastal safety network",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onPrimary
                    )
                }
            }
            
            // Main content
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .verticalScroll(scrollState)
                    .padding(horizontal = 30.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Spacer(modifier = Modifier.height(32.dp))
                
                Text(
                    text = "Sign in to your Account",
                    style = MaterialTheme.typography.titleMedium,
                    color = MaterialTheme.colorScheme.onBackground
                )
                
                Spacer(modifier = Modifier.height(40.dp))
                
                // Username label with icon
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Default.Person,
                        contentDescription = null,
                        modifier = Modifier.size(19.dp),
                        tint = OceanColors.TextSecondary
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = "Username or Email",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onBackground
                    )
                }
                
                Spacer(modifier = Modifier.height(8.dp))
                
                // Username input field - gray background matching Figma
                OutlinedTextField(
                    value = username,
                    onValueChange = { username = it },
                    placeholder = { Text("Enter your username or email", style = MaterialTheme.typography.bodyMedium, color = OceanColors.TextHint) },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(6.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        unfocusedContainerColor = OceanColors.PlaceholderGray,
                        focusedContainerColor = OceanColors.PlaceholderGray,
                        unfocusedBorderColor = Color.Transparent,
                        focusedBorderColor = OceanColors.Primary
                    ),
                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next)
                )
                
                Spacer(modifier = Modifier.height(24.dp))
                
                // Password label with icon
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Default.Lock,
                        contentDescription = null,
                        modifier = Modifier.size(19.dp),
                        tint = OceanColors.TextSecondary
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = "Password",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onBackground
                    )
                }
                
                Spacer(modifier = Modifier.height(8.dp))
                
                // Password input field - gray background matching Figma
                OutlinedTextField(
                    value = password,
                    onValueChange = { password = it },
                    placeholder = { Text("Enter your password", style = MaterialTheme.typography.bodyMedium, color = OceanColors.TextHint) },
                    trailingIcon = {
                        IconButton(onClick = { passwordVisible = !passwordVisible }) {
                            Icon(
                                imageVector = if (passwordVisible) Icons.Default.Visibility else Icons.Default.VisibilityOff,
                                contentDescription = if (passwordVisible) "Hide" else "Show"
                            )
                        }
                    },
                    visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(6.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        unfocusedContainerColor = OceanColors.PlaceholderGray,
                        focusedContainerColor = OceanColors.PlaceholderGray,
                        unfocusedBorderColor = Color.Transparent,
                        focusedBorderColor = OceanColors.Primary
                    ),
                    keyboardOptions = KeyboardOptions(
                        keyboardType = KeyboardType.Password,
                        imeAction = ImeAction.Done
                    )
                )
                
                Spacer(modifier = Modifier.height(12.dp))
                
                // Remember me + Forget Password row
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Checkbox(
                            checked = rememberMe,
                            onCheckedChange = { rememberMe = it },
                            colors = CheckboxDefaults.colors(
                                checkedColor = OceanColors.Primary,
                                uncheckedColor = OceanColors.PlaceholderGray,
                                checkmarkColor = MaterialTheme.colorScheme.onPrimary
                            ),
                            modifier = Modifier.size(15.dp)
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                        Text(
                            text = "Remember",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onBackground
                        )
                    }
                    
                    TextButton(onClick = { }) {
                        Text(
                            text = "Forget Password ?",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onBackground
                        )
                    }
                }
                
                // Error message
                if (loginState is LoginState.Error) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Card(
                        colors = CardDefaults.cardColors(
                            containerColor = OceanColors.Danger.copy(alpha = 0.1f)
                        ),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Row(
                            modifier = Modifier.padding(12.dp),
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Icon(
                                imageVector = Icons.Default.Error,
                                contentDescription = null,
                                tint = OceanColors.Danger
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = (loginState as LoginState.Error).message,
                                color = OceanColors.Danger,
                                style = MaterialTheme.typography.bodySmall
                            )
                        }
                    }
                }
                
                Spacer(modifier = Modifier.height(28.dp))
                
                // Yellow Sign In button - matches Figma design
                Button(
                    onClick = { viewModel.login(username, password) },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(42.dp),
                    enabled = username.isNotBlank() && password.isNotBlank() && loginState !is LoginState.Loading,
                    shape = RoundedCornerShape(5.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = MaterialTheme.colorScheme.primary,
                        contentColor = MaterialTheme.colorScheme.onPrimary,
                        disabledContainerColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.5f),
                        disabledContentColor = MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.5f)
                    ),
                    elevation = ButtonDefaults.buttonElevation(defaultElevation = 2.dp)
                ) {
                    if (loginState is LoginState.Loading) {
                        CircularProgressIndicator(
                            modifier = Modifier.size(20.dp),
                            color = MaterialTheme.colorScheme.onPrimary,
                            strokeWidth = 2.dp
                        )
                    } else {
                        Text(
                            text = "Sign In",
                            style = MaterialTheme.typography.labelLarge
                        )
                    }
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Demo accounts toggle
                TextButton(
                    onClick = { showDemoAccounts = !showDemoAccounts }
                ) {
                    Icon(
                        imageVector = if (showDemoAccounts) Icons.Default.KeyboardArrowUp else Icons.Default.KeyboardArrowDown,
                        contentDescription = null
                    )
                    Spacer(modifier = Modifier.width(4.dp))
                    Text("Quick Demo Access")
                }
                
                // Demo accounts
                if (showDemoAccounts) {
                    Spacer(modifier = Modifier.height(8.dp))
                    
                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        colors = CardDefaults.cardColors(
                            containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)
                        ),
                        shape = RoundedCornerShape(8.dp)
                    ) {
                        Column(
                            modifier = Modifier.padding(16.dp)
                        ) {
                            Text(
                                text = "Demo Accounts",
                                style = MaterialTheme.typography.titleSmall,
                                fontWeight = FontWeight.SemiBold
                            )
                            
                            Spacer(modifier = Modifier.height(12.dp))
                            
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.spacedBy(8.dp)
                            ) {
                                DemoRoleButton(
                                    text = "Citizen",
                                    icon = Icons.Default.Person,
                                    color = OceanColors.Green,
                                    onClick = { viewModel.demoLogin(UserRole.PUBLIC) },
                                    modifier = Modifier.weight(1f)
                                )
                                
                                DemoRoleButton(
                                    text = "Admin",
                                    icon = Icons.Default.AdminPanelSettings,
                                    color = OceanColors.Error,
                                    onClick = { viewModel.demoLogin(UserRole.ADMIN) },
                                    modifier = Modifier.weight(1f)
                                )
                            }
                            
                            Spacer(modifier = Modifier.height(8.dp))
                            
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.spacedBy(8.dp)
                            ) {
                                DemoRoleButton(
                                    text = "Rescue",
                                    icon = Icons.Default.LocalHospital,
                                    color = OceanColors.Secondary,
                                    onClick = { viewModel.demoLogin(UserRole.RESCUE_TEAM) },
                                    modifier = Modifier.weight(1f)
                                )
                                
                                DemoRoleButton(
                                    text = "Authority",
                                    icon = Icons.Default.Shield,
                                    color = OceanColors.Orange,
                                    onClick = { viewModel.demoLogin(UserRole.AUTHORITY) },
                                    modifier = Modifier.weight(1f)
                                )
                            }
                        }
                    }
                }
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // Register link
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = "Don't have an account?",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    TextButton(onClick = onNavigateToRegister) {
                        Text("Sign Up", color = OceanColors.Secondary)
                    }
                }
                
                Spacer(modifier = Modifier.height(32.dp))
            }
        }
    }
}

File: IOS/OceanSentinels/Presentation/UI/Screens/Auth/LoginScreen.swift
import SwiftUI

// MARK: - LoginScreen

/// Login screen with Ocean Sentinels Portal header, demo accounts, and form fields.
struct LoginScreen: View {
    @Environment(AuthViewModel.self) private var authViewModel
    @Environment(NavigationRouter.self) private var router

    @State private var username = ""
    @State private var password = ""
    @State private var rememberMe = false
    @State private var showDemoAccounts = false

    var body: some View {
        @Bindable var authVM = authViewModel

        VStack(spacing: 0) {
            // Yellow header
            VStack(spacing: 4) {
                Text("Ocean Sentinels Portal")
                    .font(.title2.weight(.semibold))
                Text("Secure access to India's coastal safety network")
                    .font(.subheadline)
            }
            .foregroundStyle(.white)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 20)
            .padding(.horizontal, 24)
            .background(Color.oceanPrimary)

            // Form content
            ScrollView {
                VStack(spacing: 0) {
                    Text("Sign in to your Account")
                        .font(.headline)
                        .padding(.top, 32)

                    // Username field
                    VStack(alignment: .leading, spacing: 8) {
                        Label("Username or Email", systemImage: "person.fill")
                            .font(.subheadline)
                            .foregroundStyle(.secondary)

                        TextField("Enter your username or email", text: $username)
                            #if os(iOS)
                            .textContentType(.username)
                            .textInputAutocapitalization(.never)
                            #endif
                            .autocorrectionDisabled()
                            .padding()
                            .background(Color.compatSystemGray6, in: RoundedRectangle(cornerRadius: 6))
                    }
                    .padding(.top, 40)

                    // Password field
                    VStack(alignment: .leading, spacing: 8) {
                        Label("Password", systemImage: "lock.fill")
                            .font(.subheadline)
                            .foregroundStyle(.secondary)

                        OceanPasswordField(label: "Enter your password", text: $password)
                    }
                    .padding(.top, 24)

                    // Remember me row
                    HStack {
                        Toggle(isOn: $rememberMe) {
                            Text("Remember")
                                .font(.subheadline)
                        }
                        .toggleStyle(.switch)
                        .tint(Color.oceanPrimary)

                        Spacer()

                        Button("Forgot Password?") {}
                            .font(.subheadline)
                            .foregroundStyle(.primary)
                    }
                    .padding(.top, 12)

                    // Error
                    if case .error(let message) = authVM.loginState {
                        HStack(spacing: 8) {
                            Image(systemName: "exclamationmark.circle.fill")
                            Text(message)
                                .font(.caption)
                        }
                        .foregroundStyle(Color.oceanDanger)
                        .padding(12)
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .background(Color.oceanDanger.opacity(0.1), in: RoundedRectangle(cornerRadius: 8))
                        .padding(.top, 8)
                    }

                    // Sign In button
                    OceanPrimaryButton(
                        text: "Sign In",
                        action: { authViewModel.login(username: username, password: password) },
                        isEnabled: !username.isEmpty && !password.isEmpty && authVM.loginState != .loading,
                        isLoading: authVM.loginState == .loading
                    )
                    .padding(.top, 28)

                    // Demo accounts toggle
                    Button {
                        withAnimation { showDemoAccounts.toggle() }
                    } label: {
                        HStack(spacing: 4) {
                            Image(systemName: showDemoAccounts ? "chevron.up" : "chevron.down")
                            Text("Quick Demo Access")
                        }
                        .font(.subheadline)
                    }
                    .padding(.top, 20)

                    if showDemoAccounts {
                        VStack(spacing: 12) {
                            Text("Demo Accounts")
                                .font(.subheadline.weight(.semibold))

                            LazyVGrid(columns: [
                                GridItem(.flexible()),
                                GridItem(.flexible())
                            ], spacing: 8) {
                                DemoRoleButton(text: "Citizen", icon: "person.fill", color: .oceanSuccess) {
                                    authViewModel.demoLogin(role: .public)
                                }
                                DemoRoleButton(text: "Admin", icon: "gearshape.fill", color: .oceanDanger) {
                                    authViewModel.demoLogin(role: .admin)
                                }
                                DemoRoleButton(text: "Rescue", icon: "cross.case.fill", color: .oceanPrimary) {
                                    authViewModel.demoLogin(role: .rescueTeam)
                                }
                                DemoRoleButton(text: "Authority", icon: "shield.fill", color: .orange) {
                                    authViewModel.demoLogin(role: .authority)
                                }
                            }
                        }
                        .padding(16)
                        .background(Color.compatSystemGray6.opacity(0.5), in: RoundedRectangle(cornerRadius: 8))
                        .padding(.top, 8)
                    }

                    // Register link
                    HStack {
                        Text("Don't have an account?")
                            .foregroundStyle(.secondary)
                        Button("Sign Up") {
                            router.authPath.append(AuthRoute.register)
                        }
                        .foregroundStyle(Color.oceanSecondary)
                    }
                    .font(.subheadline)
                    .padding(.top, 32)
                    .padding(.bottom, 32)
                }
                .padding(.horizontal, 30)
            }
        }
    }
}
